packages/affiliates/libs/affiliates/dappmining_v2.js

```diff
@@ -0,0 +1,278 @@
+    // transactions do not contain transactions called from other contracts! Trace does.
+    const stream = queries.streamTracesByContract(empAddress, start, end);
+    const decodeTx = DecodeTransaction(empAbi);
+
+    return (
+      highland(stream)
+        // trace transactions may contain the contract creation event, we cant decode this so filter out
+        // fyi non contract creation "trace_type" are called "call"
+        .filter(tx => tx.trace_type !== "create" && tx.error == null)
+        .map(tx => {
+          return decodeTx(tx, {
+            blockNumber: tx.block_number,
+            blockTimestamp: moment(tx.block_timestamp.value).valueOf(),
+            ...tx
+          });
+        })
+        // this filters only emp contract "create" calls. We could technically look at more, but these
+        // would be the only "attributed" calls as per the spec
+        .filter(tx => tx.name == "create")
+    );
+  }
+
+  // returns the proportion of rewards given to each developer for a specific point in time
+  function calculateBlockReward(
+    { attributions, balances, whitelist },
+    // return all tags with a percent value for rewards
+    rewards = {}
+  ) {
+    // for every user with a token balance
+    balances.keys().forEach(userid => {
+      // get users balance
+      const balance = balances.get(userid);
+      const attributionPercents = attributions.getAttributionPercents(userid, balance);
+      // go through all whitelisted tags
+      whitelist.forEach(tag => {
+        if (rewards[tag] == null) rewards[tag] = toBN("0");
+        // get that tags attribution percent for that user relative to all other affiliates
+        const attributionPercent = attributionPercents[tag] || "0";
+        // multiply the balance and the attribution percent to get final attribution weight
+        // and add it to the rest of the weights. This will need to be divided by total balances eventually.
+        rewards[tag] = rewards[tag].add(toBN(balance).mul(toBN(attributionPercent)));
+      });
+    });
+    // normalize balances based on total in all balances. This is done last instead of in the above to
+    // reduce precision loss which seems to happen the more divisions there are.
+    const total = toBN(balances.getTotal());
+    return Object.entries(rewards).reduce((result, [key, value]) => {
+      result[key] = value.div(total).toString();
+      return result;
+    }, {});
+  }
+
+  // integer percent calculation where percent is passed in as wei
+  function calculatePercent(amount, percent) {
+    return fromWei(toBN(amount.toString()).mul(toBN(percent.toString())));
+  }
+
+  // calculates attribution weights for multiple blocks by simply taking last snapshot state and multiplying by
+  // the number of blocks elapsed.
+  function sumAttributions(attributions, blocksElapsed = 1, sum = AttributionLookback()) {
+    blocksElapsed = toBN(blocksElapsed);
+    attributions.forEach((userid, affiliate, amount, index) => {
```

Comment:
this calculation changed pretty significantly, but the structure is similar to before. This updates multiplies all attributions based on number of blocks elapsed and saves them to a new lookback table.